### PR TITLE
XiaomiGateway enhancements

### PR DESCRIPTION
**Pull request by corbinmunce**
Changed wireless switch to use selector type with extra options for click/long click/double click.
Added try/catch around socket connector to handle port 9898 already bound.
Stopped WriteToHardware sending commands for wireless switch as not necessary.

### CODE REVIEW

File: hardware/XiaomiGateway.cpp [hunk 1]
```diff
@@ -112,7 +115,7 @@ void XiaomiGateway::InsertUpdateHumidity(const std::string &nodeid, const std::s
 	SendHumiditySensor(sID, 255, Humidity, Name.c_str());
 }
 
-void XiaomiGateway::InsertUpdateSwitch(const std::string &nodeid, const std::string &Name, bool bIsOn, _eSwitchType subtype)
+void XiaomiGateway::InsertUpdateSwitch(const std::string &nodeid, const std::string &Name, bool bIsOn, _eSwitchType subtype, int level)
```
Comment (gizmocuz) [inline hardware/XiaomiGateway.cpp]: Could you declares these as 'const' as well:
```
eSwitchType subtype
int level
```

File: hardware/XiaomiGateway.h [hunk 2]
```diff
@@ -13,7 +13,7 @@ class XiaomiGateway : public CDomoticzHardwareBase
 	XiaomiGateway(const int ID);
 	~XiaomiGateway(void);
 	bool WriteToHardware(const char *pdata, const unsigned char length);
-	void InsertUpdateSwitch(const std::string &nodeid, const std::string &Name, bool bIsOn, _eSwitchType subtype);
+	void InsertUpdateSwitch(const std::string &nodeid, const std::string &Name, bool bIsOn, _eSwitchType subtype, int level);
```
Comment (gizmocuz) [inline hardware/XiaomiGateway.h]: could you declare these as const also ?:
```
 _eSwitchType subtype
int level
```

File: hardware/XiaomiGateway.cpp (hunk 1 above)
Comment (corbinmunce) [inline hardware/XiaomiGateway.cpp]: sure, should bool bIsOn be a const as well?

**Comment by gizmocuz** [comment]
Yes please, anything that could be declared as const plz

**Comment by gizmocuz** [comment]
Thanks!

**Comment by corbinmunce** [comment]
You are welcome!

**Pull request merged**